crates/ruff_workspace/src/options.rs

```diff
@@ -2896,6 +2896,154 @@ pub struct FormatOptions {
         "#
     )]
     pub line_ending: Option<LineEnding>,
+
+    /// Whether to format code snippets in docstrings.
+    ///
+    /// When this is enabled, Python code in the format of doctests
```

Comment:
Nit: I'd remove "in the format of doctests", and just say, "Python code within docstrings", since this also applies to reStructuredText and Markdown.